posts/h5ad_conversion/index.qmd

```diff
@@ -0,0 +1,621 @@
+```
+![](figures/Fig-mBrain-S1-cellType-segmentation.png)
+
+You can zoom in to view a subset of cells by specifying which cells to plots.
+```{r showSubset, message=FALSE, eval = FALSE}
+# change idents to "fov" for cell selection
+SeuratObject::Idents(seu.obj1) <- "fov"
+
+Seurat::ImageDimPlot(object = seu.obj1, 
+                     fov = slideName, 
+                     
+                     # column name of cell type in meta.data
+                     group.by = "nb_clus_final", 
+                     
+                     # a vector of chosen cells, plot cells in chosen fovs
+                     cells = SeuratObject::WhichCells(
+                       seu.obj1, idents = c(72:74, 90:92, 97:99, 114:116)), 
+                     
+                     # crop the plots to area with cells only
+                     crop = TRUE, 
+                     
+                     # fixed aspect ratio and flip xy in plotting
+                     coord.fixed = TRUE, flip_xy = TRUE)
+```
+
+![](figures/Fig-mBrain-S1-cellType-fovZoomed.png)
+
+
+### Case 2: multi-slide per `anndata` object {#sec-perStudy}
+
+Since AtoMx&#8482; SIP keeps the spatial coordinates of each slide in their own spatial coordinate space, one would need to arrange the spatial coordinates of multiple sample sections to avoid overlap in XY space when exporting multiple slides in same study-level `anndata` object. The example code below would arrange all slides along Y axis with some spacing between the slides. 
+
+```{r arrangeSlides, message=FALSE, eval = FALSE}
+# extract the segmentation to separate variable
+imgList <- seu.obj@images
+
+# arrange slides along Y axis, add in spacer which is 0.2x of previous slide's coordinate span in Y direction
+spacerFactor <- 0.2
+global_y_offset <- 0 
+
+spatial_coords <- lapply(imgList, function(img){
+  # coordinates of query slide
+  eachCoord <- img$centroids@coords
+  rownames(eachCoord) <- img$centroids@cells
+  
+  # align to lower left corner
+  ori_offsets <- apply(eachCoord, 2, min)
+  eachCoord <- sweep(eachCoord, 2, ori_offsets, "-")
+  
+  # span in y direction
+  y_span <- diff(range(eachCoord[, 2]))
+
+  # add spacer in y direction 
+  eachCoord[, 2] <- eachCoord[, 2] + global_y_offset
+  
+  # update global offset for next slide
+  global_y_offset <<- global_y_offset + y_span*(1+spacerFactor)
+  
+  return(eachCoord)
+})
+spatial_coords <- do.call(rbind, spatial_coords)  
+
+# use the study-level Seurat object for downstream
+seu.obj1 <- seu.obj
+
+# prefix for file name
+slideName <- "StudyLevel" 
+
+```
+
+Let's visualize the coordinates of all cells after slide arrangement. 
+```{r vizSlides, message=FALSE, eval = FALSE}
+# add in cell type for color
+plotData <- cbind(seu.obj1[["nb_clus_final"]], 
+                  spatial_coords[colnames(seu.obj1), ])
+
+ggplot2::ggplot(plotData, 
+                # flip xy to be consistent with earlier plots
+                aes(x = y, y = x, color = as.factor(nb_clus_final))) +
```

Comment:
For consistency with how other functions are called: `ggplot2::aes(...`
I needed to add that specification when library wasn't loaded